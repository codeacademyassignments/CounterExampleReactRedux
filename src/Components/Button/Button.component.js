import './Button.component.css';
import React from 'react';
import PropTypes from 'prop-types';

const changeColor = {
  color: 'black',
};

const Button = ({ onClick, buttonText }) => {
  // console.log(onClick);
  const onClickChange = () => () => {
    onClick();
  };
  return (
    <button type="button" className="style" style={changeColor} onClick={onClickChange()}>{buttonText}</button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  buttonText: PropTypes.string.isRequired,
};

export default Button;
