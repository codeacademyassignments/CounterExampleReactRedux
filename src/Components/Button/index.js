import React, { Component } from 'react';
import './Button.css';



      class Button extends Component {
        render() {
          console.log(this.props);
          return (
            <button type="button" className = "style"  onClick={this.props.onClick}>{this.props.buttonText}</button>
            
          );
        }
      }

      export default Button;