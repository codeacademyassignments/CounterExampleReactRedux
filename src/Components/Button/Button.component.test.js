import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button.component';


describe('Button', () => {
  it('should render a button', () => {
    const tree = renderer.create(<Button buttonText="press" onClick={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should call the passed function', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Button buttonText="press" onClick={mockClick} />);
    wrapper.find('button').simulate('click');
    expect(mockClick).toHaveBeenCalled();
  });
});
