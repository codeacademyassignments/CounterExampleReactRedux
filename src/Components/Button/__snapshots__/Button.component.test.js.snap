// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button 1`] = `
<button
  className="style"
  onClick={[Function]}
  style={
    Object {
      "color": "black",
    }
  }
  type="button"
>
  press
</button>
`;
